safegraph_py_functions/safegraph_py_functions.py

```diff
@@ -225,3 +237,32 @@ def read_pattern_multi(path_to_pattern, compression='gzip', *args, **kwargs):
 def merge_core_pattern(core_df, patterns_df, how='inner', *args, **kwargs):
     merged_df = pd.merge(core_df, patterns_df, on='safegraph_place_id', how=how, *args, **kwargs)
     return merged_df
+
+
+### --------------------------------------- END CORE, GEO, AND PATTERNS SECTION -----------------------------------------------
+
+### --------------------------------------- SOCIAL DISTANCING SECTION -----------------------------------------------
+
+## start_range and end_range = string formated as "year-month-day" 
+    ## ex: start_range = "2020-06-01", end_range = "2020-06-07"
+
+def merge_socialDist_by_dates(path_to_social_dist, start_date, end_date, *args, **kwargs):
+    path = Path(path_to_social_dist,start_date[:4]) # go to year 
+    if(start_date[5:7] == end_date[5:7]): # same month
+        path = Path(path,end_date[5:7])
+        files = [file for x in path.iterdir() for file in x.iterdir()][int(start_date[-2:])-1:int(end_date[-2:])]
+    else:
+        last_day = monthrange(int(start_date[:4]), int(start_date[5:7]))[1] # get last day of month
+        path2 = Path(path, end_date[5:7]) # not same month, so different folder
+        path = Path(path, start_date[5:7])
+        files = [file for x in path.iterdir() for file in x.iterdir()][int(start_date[-2:])-1:last_day]
+        files.extend([file for x in path2.iterdir() for file in x.iterdir()][:int(end_date[-2:])])
+    t = []
+    for file in files:
+        temp_df = pd.read_csv(file,dtype= {'origin_census_block_group':str}, *args, **kwargs)
+        temp_df['origin_census_block_group']=temp_df['origin_census_block_group'].str.zfill(12) # to be safe
```

Comment:
still not sure why there is a zfill here. I saw the commit you added for it, but I am not 100% sure why it is here. Could you elaborate? 

p.s.- sorry if I am missing it

Reply:
It was just to make sure that the origin_census_block_group was correctly stored, so if it was for some reason stored and dropped a leading 0 then zfill would fix it.

However, I would rather just omit that line of code in general (which I thought I did), so if someone doesn't use origin_census_block_group then the function wont break.